goas/atomic_asset_transfer/risc0_proofs/zone_state/src/main.rs

```diff
@@ -125,59 +99,62 @@ fn deposit(
     state
 }
 
-fn validate_zone_input(
-    input: &PartialTxInputPrivate,
-    state: &StateWitness,
-) -> (PtxRoot, Nullifier) {
-    let ptx_root = input.ptx_root();
-    let nf = Nullifier::new(input.input.nf_sk, input.input.nonce);
-
-    assert_eq!(input.input.note.state, <[u8; 32]>::from(state.commit()));
-    // should not be possible to create one but let's put this check here just in case
-    debug_assert_eq!(
-        input.input.note.death_constraint,
-        state.zone_metadata.zone_vk
-    );
-
-    (ptx_root, nf)
-}
-
-fn validate_zone_output(
-    ptx: PtxRoot,
-    input: InputWitness,
-    output: PartialTxOutputPrivate,
-    state: &StateWitness,
+fn validate_zone_transition(
```

Comment:
I think we also need a way to validate the input/output notes provided here are the ones of the zone, and for example just a 1 NMO input to 1 NMO output.
Maybe check the nullifier and/or death constraint as we did before?

Reply:
We are checking the unit/death constraints are propagated here. 
Do you mean that we should be checking against the zone_metadata instead?

Reply:
Discussed offline, it's safer to code these validation w.r.t. a zone metadata